vars/util.groovy

```diff
@@ -204,4 +204,22 @@ GIT INFORMATION REPORT
     } else {
         println '[WARNING] The variable GIT_INFORMATION_REPORT does not exist'
     }
+}
+
+def getNextSnapshot(String version) {
```

Comment:
Just as a suggestion, what do you think about this @shashankcuber ?
```
def getNextVersion(String version, String type, String suffix = 'SNAPSHOT') {
    String [] versionSplit = version.split("\\.")
    if(versionSplit.length == 3) {
        if(versionSplit[0].isNumber() && versionSplit[1].isNumber() && versionSplit[2].isNumber()) { 
            int majorVersion = Integer.parseInt(versionSplit[0]) + suffix == 'major' ? 1 : 0
            int minorVersion = Integer.parseInt(versionSplit[1]) + suffix == 'minor' ? 1 : 0
            int microVersion = Integer.parseInt(versionSplit[2]) + suffix == 'micro' ? 1 : 0
            return "${majorVersion}.${minorVersion}.${microVersion+1}${suffix ? '-' + suffix : ''}"
        }
        else {
            error "Version is not in the required format.It should contain only numeric characters"
        } 
    }     
    else {
        error "Version is not in the required format X.Y.Z"
    }
}
```

Reply:
what if  `type` is an enum? :thinking:

Reply:
what do you mean by `type` as a parameter ?

I also have a problem to understand what you want to do in those lines:
```
            int majorVersion = Integer.parseInt(versionSplit[0]) + suffix == 'major' ? 1 : 0
            int minorVersion = Integer.parseInt(versionSplit[1]) + suffix == 'minor' ? 1 : 0
            int microVersion = Integer.parseInt(versionSplit[2]) + suffix == 'micro' ? 1 : 0
```
can you explain ?

Reply:
@Ginxo but as I discussed with @radtriste  version will always be passed in the format X.Y.Z so  I don't think type should be considered , but if that is the case then the one suggested by you is more clean. Can I know how a version type is decided btw?

Reply:
I'm just trying to provide a more versatile method for the rest of developers, using`type` to define which of the three part of a version will be increased (major, minor or micro). So `type` (where I recommend to use an enum) would be either `major, minor or micro`.
About this @radtriste  `int majorVersion = Integer.parseInt(versionSplit[0]) + suffix == 'major' ? 1 : 0`
`majorVersion` will be the value of `Integer.parseInt(versionSplit[0])` plus `0` or `1` depending on `type` value equal to `major` (the same for minor and micro)
@shashankcuber the value of type will be decided by the calling method, `micro` in your case

Reply:
@Ginxo ah I see now :) 
Good for me. Just one thing:
- we should use `type` instead of `suffix` 
- we should add the parenthesis for more clarity of the code

Example:
```
int majorVersion = Integer.parseInt(versionSplit[0]) + (type == 'major' ? 1 : 0`)
```

Reply:
okay I got it. Yeah that seems a more generic approach @Ginxo

Reply:
@radtriste  I am changing it and will push it. I will also change in promote pipeline by passing type as micro from my end .

Reply:
my fault there, thanks @radtriste you are right but the idea remains.